CMakeLists.txt

```diff
@@ -785,7 +785,14 @@ endif(HAVE_SIGNAL_FUNC AND HAVE_SIGNAL_MACRO)
 check_symbol_exists(uname          "${CURL_INCLUDES}" HAVE_UNAME)
 check_symbol_exists(strtoll        "${CURL_INCLUDES}" HAVE_STRTOLL)
 check_symbol_exists(_strtoi64      "${CURL_INCLUDES}" HAVE__STRTOI64)
-check_symbol_exists(strerror_r     "${CURL_INCLUDES}" HAVE_STRERROR_R)
+
+if(NOT ${CMAKE_CROSSCOMPILING})
+  # We can't tell which version of strerror_r we have unless we can run
+  # a test program; so if we are cross compiling just assume we don't have
+  # strerror_r. 
+  check_symbol_exists(strerror_r     "${CURL_INCLUDES}" HAVE_STRERROR_R)
```

Comment:
Yes, it finds the availability of the function but curl won't compile if you set HAVE_STRERROR_R but don't choose which form it is -- HAVE_GLIBC_STRERROR_R or HAVE_POSIX_STRERROR_R, which does require the try_run check. Possibly I should move this chunk down to the logic for those so it makes a bit more sense?